drivers/video/vnc/vnc_touch.c

```diff
@@ -0,0 +1,98 @@
+/****************************************************************************
+ * drivers/video/vnc/vnc_touch.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include "vnc_server.h"
+
+#include <debug.h>
+#include <errno.h>
+
+#include <nuttx/input/touchscreen.h>
+#include <nuttx/input/mouse.h>
+#include <nuttx/kmalloc.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: vnc_touch_register
+ ****************************************************************************/
+
+FAR struct touch_lowerhalf_s *vnc_touch_register(FAR const char *devpath)
+{
+  FAR struct touch_lowerhalf_s *vnc_lower;
+  int ret;
+
+  vnc_lower = kmm_zalloc(sizeof(struct touch_lowerhalf_s));
+
+  if (vnc_lower == NULL)
+    {
+      gerr("Alloc VNC touch driver instance failed.\n");
+      return NULL;
+    }
+
+  ret = touch_register(vnc_lower, devpath, 1);
+
+  if (ret < 0)
+    {
+      gerr("Register VNC touch driver failed.\n");
+      return NULL;
+    }
+
+  return vnc_lower;
+}
+
+/****************************************************************************
+ * Name: vnc_touch_event
+ ****************************************************************************/
+
+int vnc_touch_event(FAR void *arg, int16_t x,
```

Comment:
int->void

Reply:
Use int here to fit callback prototype `vnc_mouseout_t`